src/utils/offlineRequestHandler.ts

```diff
@@ -0,0 +1,348 @@
+import { useEffect } from 'react';
+import { featheryWindow } from './browser';
+
+// Constants for the IndexedDB database
+const DB_NAME = 'requestsDB';
+const STORE_NAME = 'requestsStore';
+const DB_VERSION = 1;
+const MAX_RETRY_ATTEMPTS = 3;
+const RETRY_DELAY_MS = 2000;
+
+export interface RequestOptions {
+  headers?: {
+    'Content-Type'?: string | undefined;
+    Authorization?: string;
+  };
+  method: string;
+  body: any;
+  keepalive?: boolean;
+}
+
+interface SerializedRequestBody {
+  type: 'blob' | 'formData' | 'arrayBuffer' | 'text';
+  body: ArrayBuffer | Record<string, any> | string;
+}
+
+interface SerializedRequest {
+  url: string;
+  method: string;
+  headers: string;
+  body: SerializedRequestBody['body'];
+  bodyType: SerializedRequestBody['type'];
+  timestamp: number;
+  type: string;
+  stepKey?: string;
+}
+
+export function useOfflineRequestHandler() {
+  useEffect(() => {
+    const handleOnline = () => {
+      if (!offlineRequestHandler.isReplayingRequests) {
```

Comment:
this if statement doesn't seem necessary since this guard is already in the method itself